rktl_sim/nodes/simulator

```diff
@@ -358,5 +449,244 @@ class Simulator(object):
         self.last_time = now
         self.reset_lock.release()
 
+    def get_sim_dict(self, path):
+
+        if not rospy.get_param(path):
+            raise self.InvalidRospyParam(path)
```

Comment:
the idea of raising the exception is to also print out the file path and exit the code cleanly. If you go to the IvalidRospyParam exception, you will see that it prints the exception and exits the simulation cleanly via rospy.logfatal(). Would this work, or should I replace all of my exceptions with rospy.logfatal90? I just thought that it would be nicer to print the file path that causes the error before exiting the simulation.

Reply:
Printing and logging serve different functions. Using the `print` function will hide errors when stdout is ignored by ros and doesn't store for later reference. The same info can be printed and stored using `logfatal` (see other prior examples) without requiring a need for an exception breakout.

Reply:
done!